cmd/restic/cmd_dump.go

```diff
@@ -165,24 +170,19 @@ func runDump(ctx context.Context, opts DumpOptions, gopts GlobalOptions, args []
 		return errors.Fatalf("loading tree for snapshot %q failed: %v", snapshotIDString, err)
 	}
 
-	var outputWriter io.Writer = os.Stdout
 	if opts.Target != "" {
-		file, err := os.OpenFile(opts.Target, os.O_RDWR|os.O_CREATE|os.O_EXCL, 0666)
+		file, err := os.OpenFile(opts.Target, os.O_RDWR|os.O_CREATE|os.O_EXCL, 0o666)
 		if err != nil {
 			return fmt.Errorf("cannot dump to file: %w", err)
 		}
 		defer func() {
 			_ = file.Close()
 		}()
 
-		outputWriter = file
-	} else {
-		if err := checkStdoutArchive(); err != nil {
-			return err
-		}
+		os.Stdout = file
```

Comment:
Hard reject. Overwriting such a global variable is a guarantee for problems down the road.

Please just pass a flag to `printFromTree()` that states whether to call `checkStdoutArchive()` or not. Or alternatively pass the stdout check function as a parameter, and pass a no-op function when the target is a file. The latter is probably the cleaner solution.

Reply:
Done, I have implemented the latter solution. Thank you for pointing it out. It crossed my mind.